tools/tag-release.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/sh -eu
+# SPDX-License-Identifier: GPL-2.0-or-later
+# Copyright (c) 2023 Petr Vorel <pvorel@suse.cz>
+# Tag iputils release.
+
+basedir="$(dirname "$0")"
+cd "$basedir/.."
+. "$basedir/lib.sh"
+
+upstream_git="iputils/iputils"
+tag="$(date +%Y%m%d)"
+old_tag="$(git describe --abbrev=0)"
+tag_msg="iputils-$tag"
+credit="/tmp/$(basename $upstream_git)-release-$tag"
+
+if ! git ls-remote --get-url origin | grep -q $upstream_git; then
+	quit "Not an upstream project"
+fi
+
+if ! git --no-pager diff --exit-code; then
+	quit "Please commit your changes before making new release"
```

Comment:
this doesn't handle changes staged in the index, or unknown files

Reply:
Would you add something else? Maybe
```
if ! git --no-pager diff --exit-code || ! git --no-pager diff --staged  --exit-code; then
```

Fell free to post a command. BTW I didn't meant this check to be perfect, it's just a sanity check.

Maybe not being on master branch could be a problem worth to detect (we probably don't want to release from some random branch, release is a single commit on already widely tested master branch, but merging from other branch could merge more commits to master or fail during push to master):

```
if [ "$(git branch --show-current)" != "master" ]; then
    quit "not on master branch"
fi
```
```

Reply:
personally i don't have release scripts operate in the current checkout at all.  git makes it cheap to create a new shared checkout using the current repo.  so check that out to a temp location, do the generation/commit/tag steps in there, and then throw it away.

Reply:
That's already in `tools/create-tarballs.sh`, because that's really safer for creating tarballs (to have just source files). But I thought the tag creation should be ok work in the current checkout. Sure, I can clone with `-b master`, but then I might need to ask user to verify 1) if the latest commit on master is the expected one to tag, 2) preferably then pull the release commit and tag in the working directory (not strictly needed for the release).